content/engine/security/rootless.md

```diff
@@ -581,23 +581,47 @@ $ systemctl --user restart docker
 
 **`docker run -p` does not propagate source IP addresses**
 
-This is because Docker with rootless mode uses RootlessKit's builtin port driver by default.
+This is because Docker in rootless mode uses RootlessKit's `builtin` port
+driver by default, which doesn't support source IP propagation. To enable
+source IP propagation, you can:
 
-The source IP addresses can be propagated by creating `~/.config/systemd/user/docker.service.d/override.conf` with the following content:
+- Use the `slirp4netns` RootlessKit port driver
+- Use the `pasta` RootlessKit network driver, with the `implicit` port driver
 
-```systemd
-[Service]
-Environment="DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=slirp4netns"
-```
+The `pasta` network driver is experimental, but provides improved throughput
+performance compared the `slirp4netns` port driver.
 
-And then restart the daemon:
-```console
-$ systemctl --user daemon-reload
-$ systemctl --user restart docker
-```
+To change the RootlessKit networking configuration:
+
+1. Create a file at `~/.config/systemd/user/docker.service.d/override.conf`.
+2. Add the following contents, depending on which configuration you would like to use:
+
+   - `slirp4netns`
+
+      ```systemd
+      [Service]
+      Environment="DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=slirp4netns"
+      ```
+
+   - `pasta`
+
+      ```systemd
+      [Service]
+      Environment="DOCKERD_ROOTLESS_ROOTLESSKIT_NET=pasta"
+      Environment="DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=implicit"
+      ```
+
+3. Restart the daemon:
+
+   ```console
+   $ systemctl --user daemon-reload
+   $ systemctl --user restart docker
+   ```
+
+For more information about networking options for RootlessKit, see:
 
-Note that this configuration decreases throughput.
-See [RootlessKit documentation](https://github.com/rootless-containers/rootlesskit/tree/v0.13.0#port-drivers) for the benchmark result.
+- [Network drivers](https://github.com/rootless-containers/rootlesskit/blob/v2.0.0/docs/network.md)
+- [Port drivers](https://github.com/rootless-containers/rootlesskit/blob/v2.0.0/docs/port.md)
 
```

Comment:
Can we have this table somewhere in the docs?

https://github.com/moby/moby/pull/47076/commits/b5a5ecf4a383466bbcf3d9a89dd490bc4533d803

> Network driver | Port driver    | Net throughput | Port throughput | Src IP | No SUID | Note
> ---------------|----------------|----------------|-----------------|--------|---------|--------------------------------------------
> slirp4netns    | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default in typical setup
> vpnkit         | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default when slirp4netns is not installed
> slirp4netns    | slirp4netns    | Slow           | Slow            | ✅     | ✅      |
> **pasta**      | **implicit**   | Slow           | Fast ✅         | ✅     | ✅      | Experimental; Needs recent version of pasta (`2023_12_04.b86afe3`)
> lxc-user-nic   | builtin        | Fast ✅        | Slow            | ❌     | ❌      | Experimental
> (bypass4netns) | (bypass4netns) | Fast ✅        | Fast ✅         | ✅     | ✅      | (Not integrated to RootlessKit; needs another daemon and a custom seccomp profile)